snapm/_progress.py

```
@@ -0,0 +1,336 @@
+
+        :param term_stream: The terminal stream to write to.
+        :type term_stream: ``TextIO``
+        """
+        self.header = header
+        self.term = tc or TermControl(term_stream=term_stream)
+        self.stream = term_stream or sys.stdout
+        if not (self.term.CLEAR_EOL and self.term.UP and self.term.BOL):
+            raise ValueError(
+                "Terminal isn't capable enough -- you "
+                "should use a simpler progress dispaly."
+            )
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)